### DISCUSSION TITLE
an HTTP sink + remote command?

### DISCUSSION THREAD

**Discussion by philippe44** [librespot-org/librespot#741] [Feature Requests]
Hi - I need for one application to make output data available through an HTTP server where a player can GET the audio. I assume I can do that with pipe and/or subprocess but I was thinking of adding a built-in sink which would also listen on a port where clients could GET the audio.

Optionally, that listening socket could be used to POST commands, like play/pause/stop/volume/next/previous. Could be a simple JSON payload or use an existing format like dmcp or something, I don't know. Maybe it already exists but I've missed it.

Would that be of any interest for the maintainers or shall I just keep that for myself?

**Comment by roderickvd** [comment]
Thanks for pinging back. Before entertaining this idea further, I believe it violates Spotify T&C’s 9.1 so it’s probably best not to support this.

**Reply by philippe44** [reply]
It's not the intent at all, it's just another sink because the audio data would be forever consumed by the GET. It's not like it would be available again later as a regular file, or you could do a Range request. No range, GET what's being pushed and you'd only GET it once. It does not seem different than any other sink or with subprocess(). But if you feel like it's a problem, then too bad for the idea.

**Reply by michaelherger** [reply]
I read "public distribution" in the terms. I fear that this idea is just where the grey zone, in which librespot is at home, becomes just a tad too dark...

**Reply by roderickvd** [reply]
In essence it’s broadcasting as another network service. Sure you could wire up the same with pipe, subprocess or even some loopback but then it’s clearly malicious intent of the user and not so much intent of the technology.

**Reply by michaelherger** [reply]
In the Squeezebox environment players can stream from an http source. Such a sink would allow us to pipe the audio data directly to the player. It's definitely a technical intent, not malicious. But I do understand that the librespot project tries to protect itself from too easy abuse, which is absolutely ok.

**Reply by philippe44** [reply]
So you don't think that allowing only *one* connection and only a *single* playback is fitting in the T&C? Because it's definitively not broadcast. As soon as you consume some of it, the source goes away, you can't have multiple clients and you can't play it twice.

**Reply by roderickvd** [reply]
It's not up to me (I'm no maintainer). And while I understand your point, I'd advise that it's more about not drawing unwanted attention than being right or wrong. Onlookers may not dig that deep before passing judgement.

Also, there is more to trip over in the T&C's than just broadcasting, like transferring. 

I am sorry if this curbs your enthusiasm, I don't intend to on your endeavors. I have no reason to doubt your honest intentions. Just advising caution to safeguard this project.

**Reply by philippe44** [reply]
understood

**Reply by philippe44** [reply]
So it's working now, in a purposely read-and-consume mode. At least it was a good Rust coding exercise
